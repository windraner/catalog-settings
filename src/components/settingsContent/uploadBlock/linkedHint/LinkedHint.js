import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './linkedHint.css';

export default class LinkedHint extends Component {
  render() {
    const { link } = this.props;

    return (
      <div className="catalog-settings-content__hint">
        Find more info in our guide <a href={link} className="catalog-settings-content__hint-link">here</a>.
      </div>
    );
  }
}

LinkedHint.propTypes = {
  link: PropTypes.string.isRequired,
};
