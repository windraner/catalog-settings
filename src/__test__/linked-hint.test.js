import React from 'react';
import LinkedHint from '../components/settings-content/upload-block/linkedHint/linked-hint';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<LinkedHint />', () => {
  it('Check for href', () => {
    const wrapper = shallow(<LinkedHint link="test" />);
    expect(wrapper.find('.catalog-settings-content__hint-link').props().href).toBe('test');
  });
  it('matches the snapshot', () => {
    const tree = shallow(<LinkedHint link="test" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
