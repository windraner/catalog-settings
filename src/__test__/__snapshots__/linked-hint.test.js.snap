// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkedHint /> matches the snapshot 1`] = `
<div
  className="catalog-settings-content__hint"
>
  Find more info in our guide 
  <a
    className="catalog-settings-content__hint-link"
    href="test"
  >
    here
  </a>
  .
</div>
`;
